(test) O3-4868: Add e2e test for editing a medication dispense event

gitcliff opened the pull request:
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
Add automated testing for editing a dispense event in the pharmacy app. This test will ensure that a dispense event can be edited if it was created in error. 
## Screenshots
<!-- Required if you are making UI changes. -->

## Related Issue
https://openmrs.atlassian.net/browse/O3-4868

## Other
<!-- Anything not covered above -->

denniskigen commented on e2e/specs/edit-dispense-event.spec.ts:
```diff
@@ -0,0 +1,82 @@
+test('Edit medication dispense', async ({ fhirApi, page, patient }) => {
+  const dispensingPage = new DispensingPage(page);
+
+  await test.step('Given I am on the dispensing page', async () => {
```
```suggestion
  await test.step('When I navigate to the dispensing app', async () => {
```

denniskigen committed:
(test) O3-4868: Add e2e test for editing a medication dispense event

Add an e2e test that verifies a dispense event can be edited from the
History and comments panel, covering the flow from expanding a
prescription row through updating the dispensed quantity and confirming
the updated quantity in the UI.

To support this, add a worker-scoped fhirApi Playwright fixture bound to
the FHIR R4 API, plus commands for creating and deleting a
MedicationDispense. The beforeEach hook polls the same FHIR search the
prescription details view issues until the dispense is queryable,
instead of sleeping for a fixed period.

Rebased onto main. Earlier revisions of this branch also reworded the
existing specs, enabled the Playwright and Testing Library ESLint
plugins, renamed E2E_LOGIN_DEFAULT_LOCATION_UUID, and adjusted two unit
tests. Those changes were dropped: main's specs have since been improved
in ways that supersede the rewording, the unit test suite moved to
Vitest, and the lint plugin rollout belongs in its own PR.

denniskigen committed:
Scope prescription row selection to the test patient

The e2e suite runs with parallel workers, so the All prescriptions
table contains rows created by other concurrently running specs. The
CI failure trace showed the test expanding another spec's row (whose
dispense history shows quantity 1) while its own patient's row stayed
collapsed. Select the row by the test patient's name instead of taking
the first expandable row, mirroring the approach used in the active
prescriptions spec, and scope the dispense quantity assertions to the
History and comments tab panel.

denniskigen reviewed (approved):
Thanks @gitcliff for getting this started and iterating on it. The edit-dispense flow is a real coverage gap and the FHIR fixture and dispense commands will be useful for future dispensing tests. Rebased onto main with the row selection scoped to the test patient; three consecutive green E2E runs. Merging.
